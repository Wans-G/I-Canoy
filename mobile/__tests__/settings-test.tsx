// __tests__/SettingsScreen.test.tsx

import React from 'react';
import { render } from '@testing-library/react-native';
import SettingsScreen from '../app/settings';
import { TouchableOpacity } from 'react-native';

describe('<SettingsScreen />', () => {
  let screen: ReturnType<typeof render>;

  beforeAll(() => {
    // If you get asset errors, add in your Jest setup:
    // jest.mock('@/assets/images/back-arrow.png', () => 1);
  });

  beforeEach(() => {
    screen = render(<SettingsScreen />);
  });

  it('renders the header with back button and title', () => {
    const { getByText, UNSAFE_getAllByType } = screen;
    expect(getByText('Settings')).toBeTruthy();

    // back button is a TouchableOpacity
    const tapps = UNSAFE_getAllByType(TouchableOpacity);
    // at least one TouchableOpacity (the back arrow)
    expect(tapps.length).toBeGreaterThanOrEqual(1);
  });

  it('shows personal information section with correct name & email', () => {
    const { getByText } = screen;
    expect(getByText('Personal Information')).toBeTruthy();
    expect(getByText('Name')).toBeTruthy();
    expect(getByText('Bruno Pham')).toBeTruthy();
    expect(getByText('Email')).toBeTruthy();
    expect(getByText('bruno203@gmail.com')).toBeTruthy();
  });

  it('lists all preference options', () => {
    const { getByText } = screen;
    ['Sales', 'New Arrivals', 'Delivery Status'].forEach(label =>
      expect(getByText(label)).toBeTruthy()
    );
  });

  it('lists all help center options as touchables', () => {
    const { UNSAFE_getAllByType, getByText } = screen;
    // back button + 3 help items
    const allTaps = UNSAFE_getAllByType(TouchableOpacity);
    expect(allTaps).toHaveLength(4);

    ['FAQ', 'Contact Us', 'Privacy & Terms'].forEach(label =>
      expect(getByText(label)).toBeTruthy()
    );
  });

  it('matches snapshot', () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
