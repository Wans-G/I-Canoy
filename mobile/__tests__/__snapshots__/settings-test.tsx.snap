// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsScreen /> matches snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignSelf": "center",
        "backgroundColor": "#ffffff",
        "width": 375,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 12,
          "paddingHorizontal": 16,
          "paddingTop": 20,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "height": 20,
            "opacity": 1,
            "width": 20,
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            {
              "height": 20,
              "width": 20,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../assets/images/back-arrow.png",
              }
            }
            style={
              [
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                {
                  "height": 20,
                  "width": 20,
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
      <Text
        numberOfLines={1}
        style={
          {
            "color": "#303030",
            "flex": 1,
            "fontFamily": "Merriweather",
            "fontSize": 16,
            "fontWeight": "700",
            "textAlign": "center",
          }
        }
      >
        Settings
      </Text>
      <View
        style={
          {
            "width": 20,
          }
        }
      />
    </View>
    <RCTScrollView
      contentContainerStyle={
        {
          "paddingBottom": 20,
        }
      }
    >
      <View>
        <Text
          style={
            {
              "color": "#909191",
              "fontFamily": "Nunito Sans",
              "fontSize": 16,
              "fontWeight": "600",
              "marginLeft": 20,
              "marginTop": 20,
            }
          }
        >
          Personal Information
        </Text>
        <View
          style={
            [
              {
                "backgroundColor": "#fff",
                "borderRadius": 4,
                "elevation": 2,
                "marginLeft": 20,
                "padding": 16,
                "shadowColor": "#000",
                "shadowOpacity": 0.05,
                "shadowRadius": 4,
                "width": 335,
              },
              {
                "marginTop": 10,
              },
            ]
          }
        >
          <Text
            style={
              {
                "color": "#808080",
                "fontSize": 12,
              }
            }
          >
            Name
          </Text>
          <Text
            style={
              {
                "color": "#232323",
                "fontSize": 14,
                "fontWeight": "600",
                "marginTop": 4,
              }
            }
          >
            Bruno Pham
          </Text>
        </View>
        <View
          style={
            [
              {
                "backgroundColor": "#fff",
                "borderRadius": 4,
                "elevation": 2,
                "marginLeft": 20,
                "padding": 16,
                "shadowColor": "#000",
                "shadowOpacity": 0.05,
                "shadowRadius": 4,
                "width": 335,
              },
              {
                "marginTop": 15,
              },
            ]
          }
        >
          <Text
            style={
              {
                "color": "#808080",
                "fontSize": 12,
              }
            }
          >
            Email
          </Text>
          <Text
            style={
              {
                "color": "#232323",
                "fontSize": 14,
                "fontWeight": "600",
                "marginTop": 4,
              }
            }
          >
            bruno203@gmail.com
          </Text>
        </View>
        <Text
          style={
            {
              "color": "#909191",
              "fontFamily": "Nunito Sans",
              "fontSize": 16,
              "fontWeight": "600",
              "marginLeft": 20,
              "marginTop": 30,
            }
          }
        >
          Preferences
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 4,
              "elevation": 1,
              "flexDirection": "row",
              "height": 54,
              "justifyContent": "space-between",
              "marginLeft": 20,
              "marginTop": 10,
              "paddingHorizontal": 16,
              "shadowColor": "#000",
              "shadowOpacity": 0.05,
              "shadowRadius": 4,
              "width": 335,
            }
          }
        >
          <Text
            style={
              {
                "color": "#232323",
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
          >
            Sales
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 4,
              "elevation": 1,
              "flexDirection": "row",
              "height": 54,
              "justifyContent": "space-between",
              "marginLeft": 20,
              "marginTop": 10,
              "paddingHorizontal": 16,
              "shadowColor": "#000",
              "shadowOpacity": 0.05,
              "shadowRadius": 4,
              "width": 335,
            }
          }
        >
          <Text
            style={
              {
                "color": "#232323",
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
          >
            New Arrivals
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 4,
              "elevation": 1,
              "flexDirection": "row",
              "height": 54,
              "justifyContent": "space-between",
              "marginLeft": 20,
              "marginTop": 10,
              "paddingHorizontal": 16,
              "shadowColor": "#000",
              "shadowOpacity": 0.05,
              "shadowRadius": 4,
              "width": 335,
            }
          }
        >
          <Text
            style={
              {
                "color": "#232323",
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
          >
            Delivery Status
          </Text>
        </View>
        <Text
          style={
            {
              "color": "#909191",
              "fontFamily": "Nunito Sans",
              "fontSize": 16,
              "fontWeight": "600",
              "marginLeft": 20,
              "marginTop": 30,
            }
          }
        >
          Help Center
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 4,
              "elevation": 1,
              "flexDirection": "row",
              "height": 54,
              "justifyContent": "space-between",
              "marginLeft": 20,
              "marginTop": 10,
              "opacity": 1,
              "paddingHorizontal": 16,
              "shadowColor": "#000",
              "shadowOpacity": 0.05,
              "shadowRadius": 4,
              "width": 335,
            }
          }
        >
          <Text
            style={
              {
                "color": "#232323",
                "fontSize": 16,
                "fontWeight": "700",
              }
            }
          >
            FAQ
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 4,
              "elevation": 1,
              "flexDirection": "row",
              "height": 54,
              "justifyContent": "space-between",
              "marginLeft": 20,
              "marginTop": 10,
              "opacity": 1,
              "paddingHorizontal": 16,
              "shadowColor": "#000",
              "shadowOpacity": 0.05,
              "shadowRadius": 4,
              "width": 335,
            }
          }
        >
          <Text
            style={
              {
                "color": "#232323",
                "fontSize": 16,
                "fontWeight": "700",
              }
            }
          >
            Contact Us
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 4,
              "elevation": 1,
              "flexDirection": "row",
              "height": 54,
              "justifyContent": "space-between",
              "marginLeft": 20,
              "marginTop": 10,
              "opacity": 1,
              "paddingHorizontal": 16,
              "shadowColor": "#000",
              "shadowOpacity": 0.05,
              "shadowRadius": 4,
              "width": 335,
            }
          }
        >
          <Text
            style={
              {
                "color": "#232323",
                "fontSize": 16,
                "fontWeight": "700",
              }
            }
          >
            Privacy & Terms
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
